examples/cpp_api/using_tiledb_stats.cc

```diff
@@ -89,6 +89,34 @@ void read_array() {
   Stats::enable();
   query.submit();
   Stats::dump(stdout);
+
+  // Check stats.
+  std::string stats;
+  Stats::dump(&stats);
+  if (stats.find("\"Context.StorageManager.subSubarray.add_range\": 2") ==
+      std::string::npos) {
+    throw std::logic_error("Invalid counter for add_range");
+  }
+
+  // Ensure additional calls to Query::submit have no effect on the stats
+  query.submit();
```

Comment:
I asked for this test because I thought that the stat would be added in query submit. I actually like this design better (minus a small change see my comment below) so this test can be removed. One test that I would like it to make sure that the number of ranges stays correct over a serialized query. At the moment, the deserialization code uses Subarray::set_ranges_for_dim, which doesn't call add_range and won't mess up the counters but I'd like to have a test to make sure this stays true.